mitosheet/mitosheet/parser.py

```diff
@@ -256,69 +422,113 @@ def find_column_headers(match: Any) -> Any:
             # If this column header was already covered by another column header
             # that has been found, then this column header is just a substring
             # of another column header, so we avoid matching it
-            if match_covered_by_matches([match for _, match in column_header_match_tuples], match):
+            if match_covered_by_matches([match['match_range'] for match in parser_matches], match_range):
                 return found_column_header
 
-            # Do not replace if it is part of a function, which means it has
-            # another ascii character before or after it. Or if it is part of
-            # a number. Or if it has a ( after it, then it's a function call
-            
-            if (start - 1 >= 0 and formula[start - 1].isalnum()) or \
-                (end < len(formula) and (formula[end].isalnum() or formula[end] == '(')):
-                return found_column_header
+            # First, we check if it's an unqualified column header with no index
+            if is_no_index_after_column_header_match(formula, index, start, end):
+                parser_matches.append({
+                    'type': COLUMN_HEADER_MATCH_TYPE,
+                    'match_range': match_range,
+                    'parsed': column_header,
+                    'unparsed': found_column_header,
+                    'row_offset': 0
+                })
+                return
+
+            # Second, check if column header is follwed by an index of any variety
+            number_index_label_match = get_index_match_from_number_index(formula, formula_label, index, end)
+            datetime_index_label_match = get_index_match_from_datetime_index(formula, formula_label, index, end)
+            string_index_label_match = get_index_match_from_string_index(formula, formula_label, index, end)
+
+            index_label_match = number_index_label_match or datetime_index_label_match or string_index_label_match or None
+            if index_label_match is not None:
+                parser_matches.append({
+                    'type': COLUMN_HEADER_MATCH_TYPE,
+                    'match_range': match_range,
+                    'parsed': column_header,
+                    'unparsed': found_column_header,
+                    'row_offset': index_label_match['row_offset']
+                })
+                parser_matches.append(index_label_match)
```

Comment:
Since `COLUMN_HEADER_MATCH_TYPE` either has a row_offset of 0 or is followed by a `INDEX_LABEL_MATCH_TYPE`, we could remove the `row_offset` from the `COLUMN_HEADER_MATCH_TYPE` match type all together.

Reply:
I originally had something similar this, but you need to figure out _where_ the column headers are. So you need to reconstruct these matches in either case. 

Duplicating the row_offset in both cases is a little wacky, but in practice working with this API has led to me _always_ getting exactly what I need. The API is very pleasant to consume, so I think it makes sense to keep!